lib/matplotlib/backends/backend_gtk3.py

```diff
@@ -413,16 +414,22 @@ def __init__(self, canvas, num):
         self.canvas.show()
 
         self.vbox.pack_start(self.canvas, True, True, 0)
+        # calculate size for window
+        w = int (self.canvas.figure.bbox.width)
+        h = int (self.canvas.figure.bbox.height)
 
         self.navigation = self._get_navigation()
         self.toolbar = self._get_toolbar()
+        self.statusbar = None
         if matplotlib.rcParams['toolbar'] == 'navigation':
             self.navigation.add_tools(tools)
             self.toolbar.add_tools(toolbar_tools)
+            self.statusbar = StatusbarGTK3(self.navigation)
 
-        # calculate size for window
-        w = int (self.canvas.figure.bbox.width)
-        h = int (self.canvas.figure.bbox.height)
+            self.statusbar.show()
+            self.vbox.pack_end(self.statusbar, False, False, 0)
+            size_request = self.statusbar.size_request()
+            h += size_request.height
```

Comment:
Add a separator between the toolbar and statusbar, just to make it look more conventional.